aten/src/ATen/native/BinaryOps.cpp

```diff
@@ -213,29 +215,22 @@ Tensor& true_divide_(Tensor& self, const Tensor& divisor) {
   return native::true_divide_out(self, self, divisor);
 }
 
+// NOTE: integer floor division != C++ integer division. Integer division in
+//   C++ is equivalent to torch.trunc(torch.true_divide(a, b)), while
+//   floor division is equivalent to torch.floor(torch.true_divide(a, b)).
+//   For example, in Python -5 // 2 = -3, while in C++ -5 / 2 = -2.
+// NOTE: floor divide, like Python and NumPy's floor divide, participates in
+//   type promotion.
 Tensor& floor_divide_out(Tensor& result, const Tensor& self, const Tensor& other) {
-  auto iter = TensorIterator::binary_op(result, self, other);
-  div_stub(iter.device_type(), iter);
-
-  if (result.is_floating_point()) {
-    result.trunc_();
-  }
-
+  const auto r = self.true_divide(other).floor_();
```

Comment:
does `at::true_divide_out(result, self, other).floor_()` not work?  So you don't need to copy.

Reply:
The type of result may not be copyable to by the result type of true divide out. This could be tested for, however.